### PR TITLE
Remove faulty qubits and gates sections from transpile()

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit removes the "faulty qubits" support from the transpile(). Previously the transpile() function would attempt to filter out any faulty qubits or gates reported in a BackendV1's BackendProperties object and remap the qubits based on that. However, this code was never actually functioning as the first BackendV1 based backend to attempt to set these flags is causing an internal panic in rustworkx trying to created the remapped coupling map. This code path was mildly controversial at the time of introduction (and was already reverted once in the past) and since it provably isn't valid anymore (and I wasn't clear it ever functioned as expected) as the only backend to leverage the feature crashes transpile() this commit just opts to remove it as a potential breaking change.

### Details and comments

### CODE REVIEW

**Review by jakelishman — COMMENTED** [review]
There's also a couple of minor references to faultiness in comments in `DenseLayout` and VF2's `build_average_error_map` that we might want to squash.  For completeness for others: there's also test files `test/python/providers/{faulty_backends.py,test_faulty_backend.py}`, but those are testing the `BackendV1` model not `transpile` so this PR needn't remove them.

---

Officially this PR isn't _entirely_ within our deprecation policy because we haven't issued any warning before removing this implicit behaviour from the transpiler, but we judged the impact to be minimal as we're not aware of any current backends using this feature at all and in practice the handling doesn't actually work, so we don't think anybody _can_ be using it.  Offline, @1ucian0 (one of the original authors of the feature) also signed off on this removal.

File: releasenotes/notes/remove-faulty-qubits-support-00850f69185c365e.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    When running the :func:`~.transpile` function with a :class:`~.BackendV1`
+    based backend or a :class:`~.BackendProperties` via the ``backend_properties``
+    keyword argument that has any qubits or gates flagged as faulty the function
+    will no longer try to automatically remap the qubits based on this information.
+    The method by which :func:`~.transpile` attempted to do this remapping was
+    fundamentally flawed and in most cases of such a backend it would result
+    an internal error being raised. In practice very few backends ever set the
+    fields in :class:`~.BackendProperties` to flag a qubit or gate as faulty.
+    If you were somehow successfully relying on :func:`~.transpile to do this
+    re-mapping for you you will now need to manually do that and pass a
```
[the remark this answers is not in the document]
Comment (mtreinish) [reply releasenotes/notes/remove-faulty-qubits-support-00850f69185c365e.yaml]: Fixed in: https://github.com/Qiskit/qiskit-terra/pull/9900/commits/c27f83d01fa88fb8b2286c39c475c81a2df17b94

**Comment by mtreinish** [comment]
I think after #9911 and #9840 merge I think we should update the release note to point people to using `BackendV2Converter` with the filter flag as an alternative path.

**Comment by 1ucian0** [comment]
confirming. Let's remove it.

**Review by 1ucian0 — APPROVED** [review]
Besides https://github.com/Qiskit/qiskit-terra/pull/9900#issuecomment-1505463704 (that can be in a follow up, if necessary), LGTM

**Comment by mtreinish** [comment]
The second two references need to stay: `backendproperties` is the model object that defines the payload format and the `backend_compat` file supports filtering a target to not include them as an alternative to this code. The dense layout reference should be removed in the patch series ending in #9927. The `vf2_utils` comment I think we can either leave in place or update in a followup. It's just explaining why there is an if statement guard for a condition which should be impossible.

**Review by jakelishman — APPROVED** [review]

**Comment by mtreinish** [comment]
I went ahead and enqueued this for merging because of those remaining references only 1 arguably needs to be update, but it's just a code comment

**Pull request merged**